accounts-bench/src/main.rs

```diff
@@ -69,12 +69,13 @@ fn main() {
     if fs::remove_dir_all(path.clone()).is_err() {
         println!("Warning: Couldn't remove {path:?}");
     }
-    let accounts = Accounts::new_with_config_for_benches(
+    let accounts_db = AccountsDb::new_with_config_for_benches(
         vec![path],
         &ClusterType::Testnet,
         AccountSecondaryIndexes::default(),
         AccountShrinkThreshold::default(),
     );
+    let accounts = Accounts::new(Arc::new(accounts_db));
```

Comment:
Almost all of the cases you've touched look something like this one:
```rust
let accounts = Accounts::new(Arc::new(accounts_db))
```
Wondering if we should make `Accounts::new()` take an `AccountsDb` instead of `Arc<AccountsDb>`, and then we can shift the repeated `Arc::new()` inside `Accounts::new()`?

Also, @apfitzge  had a clever suggestion on one of my PR's the other day that might be applicable here if we want to accept `Arc<AccountsDb>` or `AccountsDb`: https://github.com/solana-labs/solana/pull/34449#discussion_r1426009121

Reply:
> I'll do basically anything to avoid adding more constructors heh.

😆 😆 😆 

I'm a little on the fence as well ha and don't have too strong of an option; I just saw the repeated code so figured I'd mention it.

That being said, I think this PR is still valuable as-is; `Arc::new()` is pretty simple and having some of these repeated is probably better than someone having to choose from a handful of constructors. So, I think we could defer a decision to another PR and ship this one as-is